src/plugins/chart_expressions/expression_heatmap/common/expression_functions/heatmap_function.ts

```diff
@@ -0,0 +1,201 @@
+import { HeatmapExpressionFunctionDefinition } from '../types';
+import {
+  EXPRESSION_HEATMAP_NAME,
+  EXPRESSION_HEATMAP_GRID_NAME,
+  EXPRESSION_HEATMAP_LEGEND_NAME,
+} from '../constants';
+
+const convertToVisDimension = (columns: DatatableColumn[], accessor: string) => {
+  const column = columns.find((c) => c.id === accessor);
+  if (!column) return;
+  return {
+    accessor: column.id,
+    format: column.meta.params,
+    type: 'vis_dimension',
+  } as unknown as ExpressionValueVisDimension;
+};
+
+const prepareHeatmapLogTable = (
+  columns: DatatableColumn[],
+  accessor: string | ExpressionValueVisDimension,
+  table: Dimension[],
+  label: string
+) => {
+  const dimension =
+    typeof accessor === 'string' ? convertToVisDimension(columns, accessor) : accessor;
+  if (dimension) {
+    table.push([[dimension], label]);
+  }
+};
+
+export const heatmapFunction = (): HeatmapExpressionFunctionDefinition => ({
+  name: EXPRESSION_HEATMAP_NAME,
+  type: 'render',
+  inputTypes: ['datatable'],
+  help: i18n.translate('expressionHeatmap.function.help', {
+    defaultMessage: 'Heatmap visualization',
+  }),
+  args: {
+    // used only in legacy heatmap, consider it as @deprecated
+    percentageMode: {
+      types: ['boolean'],
+      default: false,
+      help: i18n.translate('expressionHeatmap.function.percentageMode.help', {
+        defaultMessage: 'When is on, tooltip and legends appear as percentages.',
+      }),
+    },
+    palette: {
+      types: ['palette'],
+      help: i18n.translate('expressionHeatmap.function.palette.help', {
+        defaultMessage: 'Provides colors for the values, based on the bounds.',
+      }),
+    },
+    legend: {
+      types: [EXPRESSION_HEATMAP_LEGEND_NAME],
+      help: i18n.translate('expressionHeatmap.function.legendConfig.help', {
+        defaultMessage: 'Configure the chart legend.',
+      }),
+    },
+    gridConfig: {
+      types: [EXPRESSION_HEATMAP_GRID_NAME],
+      help: i18n.translate('expressionHeatmap.function.gridConfig.help', {
+        defaultMessage: 'Configure the heatmap layout.',
+      }),
+    },
+    showTooltip: {
+      types: ['boolean'],
+      help: i18n.translate('expressionHeatmap.function.args.addTooltipHelpText', {
+        defaultMessage: 'Show tooltip on hover',
+      }),
+      default: true,
+    },
+    // not supported yet
+    highlightInHover: {
+      types: ['boolean'],
+      help: i18n.translate('expressionHeatmap.function.args.highlightInHoverHelpText', {
+        defaultMessage:
+          'When this is enabled, it highlights the ranges of the same color on legend hover',
+      }),
+    },
+    lastRangeIsRightOpen: {
+      types: ['boolean'],
+      help: i18n.translate('expressionHeatmap.function.args.lastRangeIsRightOpen', {
+        defaultMessage: 'If is set to true, the last range value will be right open',
+      }),
+      default: true,
+    },
```

Comment:
❓To be discussed with the team. The purpose of this flag is to change the last range format. For Lens it is open
![image](https://user-images.githubusercontent.com/17003240/141959400-2e0e86d6-5b54-46da-9f28-94b7b4d01d90.png)

while in Viz editor is set to the maximum value
![image](https://user-images.githubusercontent.com/17003240/141959455-e6b3d1e1-3d25-4e1b-82d3-8bcf0e9a7edb.png)

Two options here:
- Support it as it is for a smoother transition from vislib (I am open to change the name, I couldn't think of a better name)
- Not support it and follow the Lens way (last range will be set to Infinite).

I don't have a strong opinion! Let's decide together :)